### ISSUE TITLE
ActionDispatch::ParamsParser::ParseError in POST /carts

### ISSUE THREAD

**Issue by thegcat** [thegcat/oskiosk#12]
## New exception in OSKiosk

**ActionDispatch::ParamsParser::ParseError** in **POST /carts**
757: unexpected token at &#x27;abcd&#x27;

[View on Bugsnag](https://bugsnag.com/kif-ev/oskiosk/errors/54c79cdfda3fd65190310017?event_id=54c79cdf95d492b03c00449b)

## Stacktrace

    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:53 - rescue in parse_formatted_parameters
    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:32 - parse_formatted_parameters
    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:23 - call

[View full stacktrace](https://bugsnag.com/kif-ev/oskiosk/errors/54c79cdfda3fd65190310017?event_id=54c79cdf95d492b03c00449b)

**Comment by thegcat** [comment]
Not relevant.

**Comment by thegcat** [comment]
## Resolved error re-occurred in OSKiosk

**ActionDispatch::ParamsParser::ParseError** in **POST /carts**
757: unexpected token at &#x27;1234&#x27;

[View on Bugsnag](https://bugsnag.com/kif-ev/oskiosk/errors/54c79cdfda3fd65190310017?event_id=54c7a217c2d5bdb43c005d26)

## Stacktrace

    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:53 - rescue in parse_formatted_parameters
    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:32 - parse_formatted_parameters
    vendor/bundle/ruby/2.0.0/gems/actionpack-4.2.0/lib/action_dispatch/middleware/params_parser.rb:23 - call

[View full stacktrace](https://bugsnag.com/kif-ev/oskiosk/errors/54c79cdfda3fd65190310017?event_id=54c7a217c2d5bdb43c005d26)

**Comment by roguescroll** [comment]
@thegcat would you know the reason why you were getting this error and what was the fix? I am getting this same error in another rails server and was wondering what is going on.

**Comment by thegcat** [comment]
@roguescroll IIRC this happens when you send something that doesn't even look like json or pretty malformed json to an endpoint expecting json.

**Comment by thegcat** [comment]
@roguescroll IIRC this happens when you send something that doesn't even look like json or pretty malformed json to an endpoint expecting json.

The parser tries to make something of it because it's in some rather lenient mode but then goes completely belly up in a rather bizarre way. I think I made a ticket about that with an example on the rails tracker.

TL;DR: This happens with incorrect json, make sure you send well-formed json to your endpoint and this should not happen.

**Comment by roguescroll** [comment]
@thegcat thanks!
